File: bridge/bridge_cross_promo.lua
local cross_promo = {}

-- Local variables
local is_visible = false

function cross_promo.get_games_list(on_success, on_failure)
	if on_success then
		on_success(nil, {})
	end
end

function cross_promo.show()
	is_visible = true
end

function cross_promo.hide()
	is_visible = false
end

function cross_promo.is_visible()
	return is_visible
end

return cross_promo

File: bridge/src/bridge_cross_promo.cpp
#if defined(DM_PLATFORM_HTML5)
#include "bridge_cross_promo.h"
#include "bridge.h"
#include "bridge_helper.h"

int bridge::crossPromo::getGamesList(lua_State* L) {
    return makeCallback(L, js_bridge_cross_promo_getGamesList, false);
}

int bridge::crossPromo::show(lua_State* L) {
    DM_LUA_STACK_CHECK(L, 0);
    js_bridge_cross_promo_show();
    return 0;
}

int bridge::crossPromo::hide(lua_State* L) {
    DM_LUA_STACK_CHECK(L, 0);
    js_bridge_cross_promo_hide();
    return 0;
}

int bridge::crossPromo::isVisible(lua_State* L) {
    return getBoolean(L, js_bridge_cross_promo_isVisible);
}

#endif
